packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -244,15 +246,24 @@ function AuthOptions({
       return displayToast(labels.auth.error.generic);
     },
   });
-  const onProfileSuccess = async () => {
+  const onProfileSuccess = async (options: { redirect?: string } = {}) => {
+    const { redirect } = options;
     const { data } = await refetchBoot();
+
     if (data.user) {
       const provider = chosenProvider || 'password';
       onSignBackLogin(data.user as LoggedUser, provider as SignBackProvider);
     }
+
     logEvent({
       event_name: AuthEventNames.SignupSuccessfully,
     });
+
+    // if redirect is set move before modal close
+    if (redirect) {
+      await router.push(redirect);
+    }
+
     onSuccessfulRegistration?.(data?.user);
     onClose?.(null, true);
```

Comment:
I've tried to follow all the references from this `onClose` up to `OnboardingContext` and it lead to function `onCloseOnboardingModal`. What is happening it that it close the modal then this method is called:

```ts

  const onCloseOnboardingModal = () => {
    if (onboardingMode === OnboardingMode.Auto) {
      logEvent({ event_name: LogEvent.OnboardingSkip });
    }

    if (!hasTriedOnboarding) {
      setHasTriedOnboarding(true);
    }

    setIsOnboarding(false);
    setSkipIntro(false);

    if (user && !alerts.filter && onFeedPageChanged.current) {
      onFeedPageChanged.current?.();
    }
  };

```

this should set all to false and call the `onFeedPageChanged.current?.();` but I cannot understand what is doing :\

Reply:
Hmm yeah it's a tricky process (way to intertwined I think)
I'm happy with this fix for now, maybe we can log a backlog item to abstract them into more separate functions (as it will need proper testing) (CC: @nensidosari what do you think?)

Reply:
Yeah lets go with that, I'll create a ticket on web board for the above to tackle when we'll have a chance, it's really become too messy.